### ISSUE TITLE
Cannot play video with specific resolution on PC 

### ISSUE THREAD

**Issue by royfir** [RenderHeads/UnityPlugin-AVProVideo#94]
**Describe the bug**

 - Unity version: 2018.2
 - AVPro Video version: 1.9.14
 - Operating system version: Win 7 & Win 10
 - Device model: i7-3370 win7 as example
 - Video specs (resolution, frame-rate, codec, file size): as below
General
Complete name                  : C:\ZYO\ebooks\books\Fucktest\Resource\576.mp4
Format                         : MPEG-4
Format profile                 : Base Media / Version 2
Codec ID                       : mp42 (mp42/isom)
File size                      : 22.8 MiB
Duration                       : 1 min 51 s
Overall bit rate mode          : Variable
Overall bit rate               : 1 718 kb/s
Encoded date                   : UTC 2019-05-29 01:37:53
Tagged date                    : UTC 2019-05-29 01:37:53
Writing application            : mp4creator 1.4.4

Video
ID                             : 1
Format                         : AVC
Format/Info                    : Advanced Video Codec
Format profile                 : Baseline@L1.3
Format settings                : 1 Ref Frames
Format settings, CABAC         : No
Format settings, Reference fra : 1 frame
Codec ID                       : avc1
Codec ID/Info                  : Advanced Video Coding
Duration                       : 1 min 51 s
Bit rate                       : 1 588 kb/s
Width                          : 720 pixels
Height                         : 480 pixels
Display aspect ratio           : 3:2
Frame rate mode                : Variable
Frame rate                     : 29.970 (29970/1000) FPS
Minimum frame rate             : 29.970 FPS
Maximum frame rate             : 29.980 FPS
Standard                       : NTSC
Color space                    : YUV
Chroma subsampling             : 4:2:0
Bit depth                      : 8 bits
Scan type                      : Progressive
Bits/(Pixel*Frame)             : 0.153
Stream size                    : 21.1 MiB (92%)
Language                       : English
Encoded date                   : UTC 2019-05-29 01:37:53
Tagged date                    : UTC 2019-05-29 01:37:53
Codec configuration box        : avcC

Audio
ID                             : 2
Format                         : AAC LC
Format/Info                    : Advanced Audio Codec Low Complexity
Codec ID                       : mp4a-40-2
Duration                       : 1 min 51 s
Source duration                : 1 min 51 s
Bit rate mode                  : Variable
Bit rate                       : 126 kb/s
Maximum bit rate               : 147 kb/s
Channel(s)                     : 2 channels
Channel layout                 : L R
Sampling rate                  : 44.1 kHz
Frame rate                     : 43.066 FPS (1024 SPF)
Compression mode               : Lossy
Stream size                    : 1.67 MiB (7%)
Source stream size             : 1.67 MiB (7%)
Language                       : English
Encoded date                   : UTC 2019-05-29 01:37:53
Tagged date                    : UTC 2019-05-29 01:37:53

**To Reproduce**
1. play video with avpro in the application
2. show black screen and only playing audio (happened on video with width 720)
3. some only play video with no sound

**Logs**
If video fails to play then it may be due to the resolution being higher than 1920x1080 which is the limitation of the Microsoft DirectShow H.264 decoder. To resolve this you can either use Windows 8 or above (and disable 'Force DirectShow' option), resize your video, use a different codec (such as Hap or DivX), or install a 3rd party H.264 decoder such as LAV Filters.

**Screenshots**
![image](https://user-images.githubusercontent.com/39687233/60232489-58fae280-98cf-11e9-9460-43a66a44c68c.png)
![image](https://user-images.githubusercontent.com/39687233/60232503-62844a80-98cf-11e9-98fd-e6f6cdf68636.png)
![image](https://user-images.githubusercontent.com/39687233/60232506-66b06800-98cf-11e9-8906-2987a5edf9de.png)
![image](https://user-images.githubusercontent.com/39687233/60230833-d91e4980-98c9-11e9-9724-3ba890c94386.png)

**Videos**
If applicable, add a copy of your video or the URL
1. https://reurl.cc/LOdj3        (only audio ouput)
2.https://reurl.cc/8an4g         (only vidio ouput)

**Comment by AndrewRH** [comment]
I had a look here using Windows 10 - both videos could play fine.  
I'm a bit confused because in your screenshot I can see the video playing in the background, and the log shows that it loads fine.

**Comment by royfir** [comment]
> I had a look here using Windows 10 - both videos could play fine.
> I'm a bit confused because in your screenshot I can see the video playing in the background, and the log shows that it loads fine.

Yes, that is because  I already installed the Lav Filters. Before that, I got the same message and  play only audio

**Comment by AndrewRH** [comment]
I see - thanks for explaining.

Could you download [GraphStudioNext](https://ci.appveyor.com/api/projects/cplussharp/graph-studio-next/artifacts/bin/graphstudionext64.exe?job=Environment%3A%20CONFIGURATION%3DRelease%2C%20PLATFORM%3Dx64
) and drag your video into there and send the screenshot? This will help us to see which filters is it trying to use to decode the video in DirectShow.

On my machine it looks like this:
![image](https://user-images.githubusercontent.com/5418126/60260800-48983700-98d2-11e9-8aa2-61be35164757.png)

This shows it is using the standard Windows codecs.
This makes me think that you may have had another codec (pack) installed on your machine that is trying to handle this instead of the Microsoft one, and failing.

Another thing to try, is add "Microsoft DTV-DVD Video Decoder" to the preferred filter list.  This will help to force the use of the known-good Microsoft decoder on Windows 7.

![image](https://user-images.githubusercontent.com/5418126/60260895-7e3d2000-98d2-11e9-9029-3e6eabc9eacb.png)

**Comment by royfir** [comment]
I tried the way you suggested but the video still not work
![image](https://user-images.githubusercontent.com/39687233/60270299-b8cea900-9922-11e9-8000-7751cd004be7.png)
My result of  GraphStudioNext is as same as yours
The screenshot is attached

**Comment by AndrewRH** [comment]
Hmm strange....

Your above log also looks strange - it loads the video but then reports "width....1920" but the video is 720?

1) Any if you try to re-encode the video (using Handbrake / FFMPEG) then does it work?
2) Do you have any other codec installed such as Haali Splitter or FFDSHOW?  These could be interfering.
3) If you build your app as 32-bit or 64-bit does it make a difference?
4) Does the video play  in the Unity editor?
5) Does the video play from any of our demo scenes?

Thanks,

**Comment by royfir** [comment]
1. I tried to re-encode it to different resolution, like 640*480 or 1920*1080, and it works normally. I also tried to change its fps and keep resolution, and it doesn't work, either. 
2. No, I tried lav filter before then uninstall it.
3.For some reason, this project needs to support 32-bit computer
4.No on win 7
5.No, it also show black screen
![image](https://user-images.githubusercontent.com/39687233/60320919-d2690280-99ad-11e9-9df1-29372eae2e19.png)

Thanks for your patience.
